docs/DeveloperGuide.md

```diff
@@ -171,6 +174,41 @@ As such, the detailed descriptions for the Address Book subsystem above can be t
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Import and Export CSV Features
+
+This section describes some of the details as to how the import and export CSV features were implemented
+
+#### Import CSV
+
+The import CSV function is meant to append to the current address book with new data imported from any CSV file. The intention is to allow users to be able to import from a Microsoft Excel compatible format. Since there are multiple different templates for contacts in CSV files across various platforms, such as Microsoft Outlook and Google Contacts, the idea was to ensure this feature was as flexible as possible, allowing the user to specify which particular columns contained certain information
+
+The arguments that are parsed here are the custom column numbers for each value, e.g. `n/3 p/4 e/5 a/6 t/7` will read `Name` from column 3, `Phone` from column 4, `Email` from column 5, `Address` from column 6 and `Tags` from column 7 
+
+In the event that any of the data read does not conform to the restrictions given by each of the components in `Person`, e.g. an email that does not have `@`, or if the person has a duplicate name with another person, that particular line will be skipped.
+
+The `CSVManager` takes in the `IndexedCsvFile` object, opens the file and reads the lines, while a function from the command is passed to the Manager that will store each of the lines as a new Person in the model. The sequence diagram is as follows:
```

Comment:
I suggest this version of the description, which better captures where the logic is.
```suggestion
The `CSVManager` takes in a `IndexedCsvFile` object, opens the file and reads the lines. The logic that performs the parsing of data fields and creation of `Person` models is specified in the import command, and is passed as an anonymous function to CSVManager. The sequence diagram is as follows:
```

Reply:
thanks for the suggesiton! implemented in https://github.com/AY2122S2-CS2103-W17-1/tp/pull/203/commits/3193b02116aeddfa74a438d365ea5575d9acfb0b